pkg/controller/elasticsearch/observer/manager.go

```diff
@@ -85,8 +92,7 @@ func (m *Manager) Observe(ctx context.Context, cluster esv1.Elasticsearch, esCli
 		// in case asynchronous observation has been disabled ensure at least one observation at reconciliation time.
 		return m.getAndObserveSynchronously(ctx, nsName)
 	default:
-		// No change, close the provided Client and return the existing observer.
-		esClient.Close()
```

Comment:
This was preventing the memory leak (https://github.com/elastic/cloud-on-k8s/pull/2916/files). Where do close al clients that are not reused because connexion information has changed?

Reply:
https://github.com/elastic/cloud-on-k8s/blob/180210427177dbf1046417e08be4a8d92c2ce2f1/pkg/controller/elasticsearch/observer/observer.go#L99 closes the client when the observer is stopped. This one only existed to close client that were created unnecessarily.

Reply:
Perfect.